src/services/healthcheck.service.js

```diff
@@ -99,14 +99,21 @@ class HealthService {
     let output;
     let healthStatus;
 
-    // TODO: We will need to parse the healthData to get the status,
-    // but hathor-core hasn't this implemented yet.
-    // Make sure we treat 'warn' as 'pass' if considerHealthcheckWarnAsUnhealthy is false
     try {
-      await healthApi.getHealth();
+      const healthData = await healthApi.getHealth();
 
-      output = 'Fullnode is responding';
-      healthStatus = HealthcheckStatus.PASS;
+      healthStatus = healthData.status;
+
+      const config = getConfig();
+
+      const isUnhealthy = healthStatus === HealthcheckStatus.FAIL
+        || (healthStatus === HealthcheckStatus.WARN && config.considerHealthcheckWarnAsUnhealthy);
+
+      if (isUnhealthy) {
+        output = `Fullnode reported as unhealthy: ${JSON.stringify(healthData)}`;
```

Comment:
chore: Since we're adding complexity because of the increased possible results of the healthcheck, I think we should also increase the coverage here.

Codecov is informing that this line was never reached by tests. Could we add it to the test suite?